src/Streams.jl

```diff
@@ -303,12 +303,23 @@ function Base.unsafe_read(http::Stream, p::Ptr{UInt8}, n::UInt)
     nothing
 end
 
+Base.readbytes!(http::Stream, buf::Base.BufferStream, n=bytesavailable(http)) = Base.readbytes!(http, buf.buffer, n)
```

Comment:
Why did you have to special case `BufferStream` and `IOBuffer`? Will this fail with other IO now?